lib/bosh/BoshDirectorClient.js

```diff
@@ -64,26 +69,54 @@ class BoshDirectorClient extends HttpClient {
   }
 
   populateCache() {
+    logger.info('Loading Bosh DeploymentName cache... current cached deployments:', _.keys(this.cache));
     this.cacheLoadInProgress = true;
     this.clearCache();
     return Promise
-      .map(config.directors, directorConfig => {
-        return this
-          .getDeploymentsByConfig(directorConfig)
-          .tap(deployments => _.map(deployments, deployment => {
-            if (this.cache[deployment.name] === undefined) {
-              this.cache[deployment.name] = directorConfig;
-            } else {
-              this.cache[deployment.name] = CONST.ERR_CODES.DEPLOYMENT_NAME_DUPED_ACROSS_DIRECTORS;
-            }
-          }));
-      })
-      .tap(() => {
+      .map(config.directors,
+        (directorConfig) => this.getDeploymentsByConfig(directorConfig))
+      .finally(() => {
         this.cacheLoadInProgress = false;
-        logger.debug('Cached Deployments:', _.keys(this.cache));
+        logger.info('Clearing cacheLoadInProgress flag. Bosh DeploymentName cache is loaded.');
+        logger.silly('Cached Deployments:', _.keys(this.cache));
       });
   }
 
+  getDeploymentNamesFromCache(boshName, attempt) {
+    return Promise.try(() => {
+      if (this.cacheLoadInProgress) {
+        if (!attempt) {
+          attempt = 1;
+        } else if (attempt > CONST.BOSH_POLL_MAX_ATTEMPTS) {
+          throw errors.Timeout.toManyAttempts(CONST.BOSH_POLL_MAX_ATTEMPTS, new Error('Fetching deployments from Cache is taking too long.'));
+        }
+        logger.info(`Cache load in progress. GetDeploymentNames will be delayed by 500 ms - current attempt ${attempt}`);
+        return Promise.delay(500 * attempt).then(() => this.getDeploymentNamesFromCache(boshName, ++attempt));
```

Comment:
500 put in a constant and use that in logger statement in above also.

Reply:
Will take it up post hotfix.